### PR TITLE
Remove more usages of LocationXY(Z)16

### PR DESCRIPTION
**Pull request by Gymnasiast**
This cuts even more usages of LocationXY(Z)16. The remaining usage is in paint calls.

This PR should not be squash merged, but merged in the conventional way.

### CODE REVIEW

File: src/openrct2-ui/windows/Map.cpp [hunk 1]
```diff
@@ -1629,8 +1629,7 @@ static void map_window_set_pixels(rct_window* w)
     int32_t x = 0, y = 0, dx = 0, dy = 0;
 
     int32_t pos = (_currentLine * (MAP_WINDOW_MAP_SIZE - 1)) + MAXIMUM_MAP_SIZE_TECHNICAL - 1;
-    LocationXY16 destinationPosition = { static_cast<int16_t>(pos % MAP_WINDOW_MAP_SIZE),
-                                         static_cast<int16_t>(pos / MAP_WINDOW_MAP_SIZE) };
+    ScreenCoordsXY destinationPosition = { pos % MAP_WINDOW_MAP_SIZE, pos / MAP_WINDOW_MAP_SIZE };
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Map.cpp]:
```suggestion
    auto destinationPosition = ScreenCoordsXY{ pos % MAP_WINDOW_MAP_SIZE, pos / MAP_WINDOW_MAP_SIZE };
```

File: src/openrct2-ui/windows/TrackDesignPlace.cpp [hunk 2]
```diff
@@ -127,9 +127,9 @@ static void window_track_place_draw_mini_preview_track(
     TrackDesign* td6, int32_t pass, CoordsXY origin, CoordsXY min, CoordsXY max);
 static void window_track_place_draw_mini_preview_maze(
     TrackDesign* td6, int32_t pass, const CoordsXY& origin, CoordsXY min, CoordsXY max);
-static LocationXY16 draw_mini_preview_get_pixel_position(int16_t x, int16_t y);
-static bool draw_mini_preview_is_pixel_in_bounds(LocationXY16 pixel);
-static uint8_t* draw_mini_preview_get_pixel_ptr(LocationXY16 pixel);
+static ScreenCoordsXY draw_mini_preview_get_pixel_position(int16_t x, int16_t y);
+static bool draw_mini_preview_is_pixel_in_bounds(ScreenCoordsXY pixel);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TrackDesignPlace.cpp]:
```suggestion
static bool draw_mini_preview_is_pixel_in_bounds(const ScreenCoordsXY& pixel);
```

File: src/openrct2-ui/windows/TrackDesignPlace.cpp [hunk 3]
```diff
@@ -127,9 +127,9 @@ static void window_track_place_draw_mini_preview_track(
...
+static uint8_t* draw_mini_preview_get_pixel_ptr(ScreenCoordsXY pixel);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TrackDesignPlace.cpp]:
```suggestion
static uint8_t* draw_mini_preview_get_pixel_ptr(const ScreenCoordsXY& pixel);
```

**Comment by Gymnasiast** [comment]
@tupaschoal Should now be ready to merge.

File: src/openrct2-ui/windows/TrackDesignPlace.cpp [hunk 4]
```diff
@@ -127,9 +127,9 @@ static void window_track_place_draw_mini_preview_track(
     TrackDesign* td6, int32_t pass, CoordsXY origin, CoordsXY min, CoordsXY max);
 static void window_track_place_draw_mini_preview_maze(
     TrackDesign* td6, int32_t pass, const CoordsXY& origin, CoordsXY min, CoordsXY max);
-static LocationXY16 draw_mini_preview_get_pixel_position(int16_t x, int16_t y);
-static bool draw_mini_preview_is_pixel_in_bounds(LocationXY16 pixel);
-static uint8_t* draw_mini_preview_get_pixel_ptr(LocationXY16 pixel);
+static ScreenCoordsXY draw_mini_preview_get_pixel_position(CoordsXY location);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TrackDesignPlace.cpp]: Make it
```suggestion
static ScreenCoordsXY draw_mini_preview_get_pixel_position(const CoordsXY& location);
```

**Pull request merged**